StudyHub/StudyHub.FluentEmail/Services/EmailService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FluentEmail.Core;
+using Microsoft.Extensions.Options;
+using StudyHub.Common.Models;
+using StudyHub.FluentEmail.MessageBase;
+using StudyHub.FluentEmail.Services.Interfaces;
+
+namespace StudyHub.FluentEmail.Services;
+
+public class EmailService : IEmailService
+{
+    private readonly IFluentEmail _fluentEmail;
+    private readonly EmailSettings _messageSettings;
+
+    public EmailService(IFluentEmail fluentEmail, IOptions<EmailSettings> messageSettings)
+    {
+        _fluentEmail = fluentEmail;
+        _messageSettings = messageSettings.Value;
+    }
+
+    public async Task<bool> SendAsync<T>(string to, T message)
+        where T : EmailMessageBase
+    {
+        var path = $@"{Directory.GetCurrentDirectory()}{_messageSettings.MessagePath}\{message.TemplateName}.cshtml";
+
+        var sendEmail = await _fluentEmail
+                  .To(to)
+                  .Subject(message.Subject)
+                  .UsingTemplateFromFile(path, message)
+                  .SendAsync();
+
+        _fluentEmail.Data.ToAddresses.Clear();
```

Comment:
remove this

Reply:
we need this, because if we send message to several emails, for instance 3, they send to 1, than to 2 emails at the same time, and than 3 at the same time

Reply:
won't _fuentEmail.Data clear itself after disposing?

Reply:
Of course we need. I think it will be easier to user send many invitation by adding amout of emails

Reply:
take a look on IFluentEmailFactory, it will help you to fix this issue

Reply:
now I can send an invitation letter to several users without the method SenderMany